### PR TITLE
fix: Upgrade to AWS JS SDK v3

### PR DESCRIPTION
**Pull request by mman**
Closes: https://github.com/parse-community/parse-server-s3-adapter/issues/197

The changes appear to be rather minimal and for now working in my staging environment.

Need a way to actually run proper tests and fix any remaining issues.

@mtrezza Could you help me understand how to run the tests? Or will they run automatically on PR?

### CODE REVIEW

**Comment by mtrezza** [comment]
@mman Did you try `npm test` locally?

**Comment by mman** [comment]
@mtrezza So I made some good progress on making `npm test` pass on my local machine talking to AWS S3. Couple issues remain:

1/ The interface for `FilesAdapter.getFileLocation` (https://github.com/mman/parse-server/blob/af72a4753967ad5c015f800830510bf928bdc107/src/Adapters/Files/FilesAdapter.js#L64) declares the method to return Absolute URL string synchronously.

However notes on migration to v3 mention that there is no sync method to generate pre-signed URL anymore (https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/migrating/notable-changes/). My JS experience is lacking in what to do now. Basically all remaining tests now fail because I was not able to make `getFileLocation` work as intended, and I am not sure I even understand what the intended behaviour in case of S3 is.

2/ Using V3 `putObject` method instead of V2 `upload` method returns different `Body` type back.

The original V2 `upload` (https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property) returns some kind of stream or Buffer?

V3 returns https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-smithy-types/TypeAlias/NodeJsRuntimeStreamingBlobPayloadOutputTypes/ which is basically a http response wrapper that can be converted to byte array or string (https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-smithy-types/TypeAlias/SdkStream/). I have for now used conversion to string, but I feel it is not appropriate.

The `FilesAdatapter` interface does not mention what `getFileData` should return, but one of the tests assumes it is instance of `Buffer`.

Need some help here to make progress.

File: spec/test.spec.js [hunk 1]
```diff
@@ -190,22 +190,23 @@ describe('S3Adapter tests', () => {
     });
 
     it('should accept endpoint as an override option in args', () => {
```
Comment (mman) [inline spec/test.spec.js:192]: Looks like the `AWS.Endpoint` API simply disappeared and I am not sure what to do here.

File: index.js [hunk 2]
```diff
@@ -181,19 +194,19 @@ class S3Adapter {
...
+    //   if (this._presignedUrlExpires) {
+    //     params.Expires = this._presignedUrlExpires;
+    //   }
+    //   // Always use the "getObject" operation, and we recommend that you protect the URL
+    //   // appropriately: https://docs.aws.amazon.com/AmazonS3/latest/dev/ShareObjectPreSignedURL.html
+    //   presignedUrl = getSignedUrl(this._s3Client, 'getObject', params).then(result => );
+    //   if (!this._baseUrl) {
+    //     return presignedUrl;
+    //   }
+    // }
```
Comment (mman) [inline index.js:210]: I have disabled this test as I do not know how to properly migrate to AWS SDK v3 which does not offer a sync version of `getSignedUrl` anymore.

File: index.js [hunk 3]
```diff
@@ -165,7 +178,7 @@ class S3Adapter {
         if (data && !data.Body) {
           return reject(data);
         }
-        return resolve(data.Body);
```
Comment (mman) [inline index.js:168]: Here I cast `data.Body` to `string` which is most probably wrong as we should pass up a stream?

**Comment by thphuc** [comment]
Hi @mman Did you get time to fix this?

**Comment by mtrezza** [comment]
See https://github.com/parse-community/parse-server-s3-adapter/pull/220

**Comment by mman** [comment]
> See #220

Much cleaner implementation, will close this PR in favor of #220

**Pull request closed without merging**